torch_np/_detail/_reductions.py

```diff
@@ -0,0 +1,165 @@
+""" Implementation of reduction operations, to be wrapped into arrays, dtypes etc
+in the 'public' layer.
+
+Anything here only deals with torch objects, e.g. "dtype" is a torch.dtype instance etc
+"""
+
+import torch
+from . import _util
+from . import _scalar_types
+
+NoValue = None
+
+
+def _atleast_float(dtype, other_dtype):
+    """Cast bools and ints and floats to a default float; complex to default complex.
```

Comment:
This line is a little misleading, it seems to be about scalar values rather than dtype objects. I propose:
```
Return a dtype that is real or complex floating-point.

For inputs that are boolean or integer dtypes, this returns the default float dtype;
inputs that are complex get converted to the default complex dtype;
real floating-point dtypes (`float*`) get passed through unchanged
```

And now that I've written that out - is it really the intended behavior here for `complex*` to be converted to `complex64` but `float*` to be passed through?

Reply:
Thanks. The last commit takes over you text + makes complex dtypes pass through intact.